src/compiler/emitter.ts

```diff
@@ -3991,63 +3999,146 @@ module ts {
                 emitEnd(node);
             }
 
-            function emitImportDeclaration(node: ImportDeclaration) {
-                var emitImportDeclaration = resolver.isReferencedImportDeclaration(node);
-
-                if (!emitImportDeclaration) {
-                    // preserve old compiler's behavior: emit 'var' for import declaration (even if we do not consider them referenced) when
-                    // - current file is not external module
-                    // - import declaration is top level and target is value imported by entity name
-                    emitImportDeclaration = !isExternalModule(currentSourceFile) && resolver.isTopLevelValueImportWithEntityName(node);
+            function emitRequire(moduleName: Expression) {
+                if (moduleName.kind === SyntaxKind.StringLiteral) {
+                    write("require(");
+                    emitStart(moduleName);
+                    emitLiteral(<LiteralExpression>moduleName);
+                    emitEnd(moduleName);
+                    emitToken(SyntaxKind.CloseParenToken, moduleName.end);
+                    write(";");
                 }
+                else {
+                    write("require();");
+                }
+            }
 
-                if (emitImportDeclaration) {
-                    if (isExternalModuleImportDeclaration(node) && node.parent.kind === SyntaxKind.SourceFile && compilerOptions.module === ModuleKind.AMD) {
-                        if (node.flags & NodeFlags.Export) {
-                            writeLine();
-                            emitLeadingComments(node);
-                            emitStart(node);
-                            emitModuleMemberName(node);
-                            write(" = ");
-                            emit(node.name);
-                            write(";");
-                            emitEnd(node);
-                            emitTrailingComments(node);
-                        }
-                    }
-                    else {
-                        writeLine();
+            function emitImportAssignment(node: Declaration, moduleName: Expression) {
+                if (!(node.flags & NodeFlags.Export)) write("var ");
+                emitModuleMemberName(<Declaration>node);
+                write(" = ");
+                emitRequire(moduleName);
+            }
+
+            function emitImportDeclaration(node: ImportDeclaration | ImportEqualsDeclaration) {
+                var info = getExternalImportInfo(node);
+                if (info) {
+                    var declarationNode = info.declarationNode;
+                    var namedImports = info.namedImports;
+                    if (compilerOptions.module !== ModuleKind.AMD) {

```

Comment:
No, because we default to emitting CommonJS style when the user hasn't opted into modules (along with giving an error).